pkg/flowaggregator/s3uploader/s3uploader_test.go

```diff
@@ -166,7 +194,16 @@ func TestBatchUploadAllError(t *testing.T) {
 }
 
 func TestFlowRecordPeriodicCommit(t *testing.T) {
-	mockS3Uploader := &mockS3Uploader{testReader: &bytes.Buffer{}}
+	ctrl := gomock.NewController(t)
+	mockS3Uploader := s3uploadertesting.NewMockS3UploaderAPI(ctrl)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	mockS3Uploader.EXPECT().Upload(context.Background(), gomock.Any(), nil).DoAndReturn(
+		func(arg0 interface{}, arg1 interface{}, arg2 interface{}, arg3 ...interface{}) (*s3manager.UploadOutput, error) {
+			wg.Done()
+			return nil, nil
+		},
```

Comment:
I think you should match the actual argument types for the sake of readability, or just use:
```go
func(args ...interface{}) (*s3manager.UploadOutput, error) {
    // ...
}
```
since we don't actually need the arguments